src/main/scala/firrtl/util/BackendCompilationUtilities.scala

```diff
@@ -24,17 +37,25 @@ trait BackendCompilationUtilities {
     out.close()
   }
 
-  /** Create a temporary directory with the prefix name. Exists here because it doesn't in Java 6.
+  /** Create a test directory
+    *
+    * Will create outer directory called testName then inner directory based on
+    * the current time
     */
-  def createTempDirectory(prefix: String): File = {
-    val temp = File.createTempFile(prefix, "")
-    if (!temp.delete()) {
-      throw new IOException(s"Unable to delete temp file '$temp'")
-    }
-    if (!temp.mkdir()) {
-      throw new IOException(s"Unable to create temp directory '$temp'")
+  def createTestDirectory(testName: String): File = {
+    val outer = new File(TestDirectory, testName)
+    val testDir = {
+      val dir: Future[File] = Future {
+        var _dir = new File(outer, timeStamp)
+        while (!_dir.mkdirs()) {
+          Thread.sleep(1)
+          _dir = new File(outer, timeStamp)
+        }
+        _dir
+      }
+      Await.result(dir, 10 seconds)
     }
```

Comment:
I thought about using that but I thought it might be nice to have the filenames be based on the current time instead of just being a bunch of random numbers. Do you think using `createTempDirectory` would be better?